plugin/plugin.go

```diff
@@ -225,17 +223,6 @@ func (c execCommander) Output(ctx context.Context, name string, command proto.Co
 	return stdout.Bytes(), nil, nil
 }
 
-// ParsePluginName checks if fileName is a valid plugin file name
-// and gets plugin name from it based on spec: https://github.com/notaryproject/specifications/blob/main/specs/plugin-extensibility.md#installation
-func ParsePluginName(fileName string) (string, error) {
```

Comment:
Isn't removing this a breaking change? Should we mark this as obsolete instead?

Reply:
@rgnote This won't be a breaking change as it's in the `main` branch ONLY. In other words, should we release the next version `v1.1.0`, removing it would become a breaking change.

Reply:
Right. Won't we release this code at some point? We will, right. Unless the next release is a major version upgrade, we should not let this into main branch.

Reply:
> Right. Won't we release this code at some point? We will, right. Unless the next release is a major version upgrade, we should not let this into main branch.

Please note, this code is not in any Notation release yet, therefore, this change is not a breaking change. /cc: @shizhMSFT

Reply:
> it won't be a breaking change when we release v1.1.0, because to v1.1.0, it's a new feature

Can you please elaborate more on this? If this change is included in v1.1.0, then any consumers of `ParsePluginName` API will break, right?

Reply:
> > it won't be a breaking change when we release v1.1.0, because to v1.1.0, it's a new feature
> 
> Can you please elaborate more on this? If this change is included in v1.1.0, then any consumers of `ParsePluginName` API will break, right?

We don't have any consumer of `ParsePluginName` because it's never been released by Notation: https://github.com/notaryproject/notation-go/releases. Therefore, users will start to use this function only after we release `v1.1.0`.

Reply:
> We don't have any consumer of ParsePluginName yet because it's never been released by Notation: https://github.com/notaryproject/notation-go/releases.

Got it. Thanks.